### PR TITLE
Update commands to add MI instances

### PR DESCRIPTION
**Pull request by SanojPunchihewa**
## Purpose
Update MI CLI tool and merge with APICTL
wso2/micro-integrator#1999

## Goals
- Update env command to add/remove MI instances
- Add new alias **mi** to add MI related commands
- Add new login command under mi to login to MI instances
- Add new logout command under mi to logout from MI instances

## User stories
- A new flag **--mi** is introduced under add env command to add a MI instance
`apictl add env dev --mi  https://localhost:9164`
- To login to MI instance
`apictl mi login <env> -u <username> -p <password>`
- To logout from MI instance
`apictl mi logout <env>`

## Test environment
Ubuntu 20.04.1 LTS
Go version go1.15 linux/amd64

### CODE REVIEW

File: import-export-cli/cmd/addEnv.go [hunk 1]
```diff
@@ -32,6 +32,7 @@ var flagDevPortalEndpoint string    // DevPortal endpoint of the environment to
 var flagRegistrationEndpoint string // registration endpoint of the environment to be added
 var flagApiManagerEndpoint string   // api manager endpoint of the environment to be added
 var flagAdminEndpoint string        // admin endpoint of the environment to be added
+var flagMiManagementEndpoint string // mi management endpoint of the environment to be added
```
Comment (wasuradananjith) [inline import-export-cli/cmd/addEnv.go:35]: Shall we run "go run tools/gen.go" to generate the docs (.md files) to the whole feature and commit those files here?
Refer [1] "Generating docs" section.

[1] https://github.com/wso2/product-apim-tooling/blob/master/import-export-cli/README.md

File: import-export-cli/cmd/addEnv.go [hunk 2]
```diff
@@ -41,12 +42,16 @@ const addEnvCmdLongDesc = "Add new environment and its related endpoints to the
 const addEnvCmdExamples = utils.ProjectName + ` ` + AddCmdLiteral + ` ` + AddEnvCmdLiteralTrimmed + ` production \
 --apim  https://localhost:9443 
 
+` + utils.ProjectName + ` ` + AddCmdLiteral + ` ` + AddEnvCmdLiteralTrimmed + ` dev \
```
Comment (Chamindu36) [inline import-export-cli/cmd/addEnv.go:45]: Shall we add an example with the minimum arguments need to provide when adding a new environment with both APIM and MI instances?

```
 utils.ProjectName + ` ` + AddCmdLiteral + ` ` + AddEnvCmdLiteralTrimmed + ` prod \
--apim  https://apim.com:9443 \
--mi https://localhost:9164
```

File: import-export-cli/cmd/addEnv.go (hunk 1 above)
Comment (SanojPunchihewa) [reply import-export-cli/cmd/addEnv.go:35]: Fixed with e053d59

File: import-export-cli/cmd/addEnv.go (hunk 2 above)
Comment (SanojPunchihewa) [reply import-export-cli/cmd/addEnv.go:45]: Fixed with 224f0cc

File: import-export-cli/cmd/addEnv.go (hunk 1 above)
Comment (wasuradananjith) [reply import-export-cli/cmd/addEnv.go:35]: Shall we add the commands and usage of this feature to the file in [2] as well?

[2] https://github.com/wso2/product-apim-tooling/blob/master/import-export-cli/resources/README.html

Comment (SanojPunchihewa) [reply import-export-cli/cmd/addEnv.go:35]: Fixed with 49921b9

**Review by Chamindu36 — APPROVED** [review]

**Review by wasuradananjith — APPROVED** [review]

**Review by uvindra — APPROVED** [review]

File: import-export-cli/cmd/login.go [hunk 3]
```diff
@@ -119,8 +119,8 @@ func runLogin(store credentials.Store, environment, username, password string) e
 		return err
 	}
 
-	fmt.Println("Logged into", environment, "environment")
-	err = store.Set(environment, username, password, clientId, clientSecret)
+	fmt.Println("Logged into APIM in", environment, "environment")
```
Comment (npamudika) [inline import-export-cli/cmd/login.go:122]: Should be corrected as fmt.Println("Logged into APIM in ", environment, "environment"), missing a space after "in"

File: import-export-cli/cmd/login.go [hunk 4]
```diff
@@ -135,16 +135,22 @@ func GetCredentials(env string) (credentials.Credential, error) {
 	if err != nil {
 		return credentials.Credential{}, err
 	}
+
+	if !utils.APIMExistsInEnv(env, utils.MainConfigFilePath) {
+		fmt.Println("APIM does not exists in", env, "Add it using add env")
```
Comment (npamudika) [inline import-export-cli/cmd/login.go:140]: Same here

File: import-export-cli/cmd/login.go [hunk 5]
```diff
@@ -135,16 +135,22 @@ func GetCredentials(env string) (credentials.Credential, error) {
...
+		os.Exit(1)
+	}
+
 	// check for creds
-	if !store.Has(env) {
-		fmt.Println("Login to", env)
+	if !store.HasAPIM(env) {
+		fmt.Println("Login to APIM in", env)
```
Comment (npamudika) [inline import-export-cli/cmd/login.go:146]: Let's modify here too

File: import-export-cli/cmd/logout.go [hunk 6]
```diff
@@ -60,8 +60,8 @@ func runLogout(environment string) error {
 	if err != nil {
 		return err
 	}
-	fmt.Println("Logged out from", environment, "environment")
-	return store.Erase(environment)
+	fmt.Println("Logged out from APIM in", environment, "environment")
```
Comment (npamudika) [inline import-export-cli/cmd/logout.go:63]: Let's add a space after "in" to improve readablity

File: import-export-cli/README.md [hunk 7]
```diff
@@ -1,7 +1,7 @@
-# CLI for Importing and Exporting APIs and Applications
+# CLI for Importing and Exporting APIs and Applications and Managing WSO2 Micro Integrator
 ## For WSO2 API Manager 3.2.0
 
-Command Line tool for importing and exporting APIs/Applications between different API Environments
+Command Line tool for importing and exporting APIs/Applications between different API Environments of WSO2 API Manager and Managing WSO2 Micro Integrator
```
Comment (npamudika) [inline import-export-cli/README.md:4]: Shall we modify like this? Please note the upper case and lower case letters as well :)

"Command Line tool for importing and exporting APIs/Applications/API Products in between different API environments of WSO2 API Manager and managing WSO2 Micro Integrator"

File: import-export-cli/cmd/login.go (hunk 3 above)
Comment (SanojPunchihewa) [reply import-export-cli/cmd/login.go:122]: When `,` is used to concatenate strings, `Println` will add a space by default. https://play.golang.org/p/0lAnwRXpXrs

Comment (npamudika) [reply import-export-cli/cmd/login.go:122]: no issues then. We can keep like this.

**Pull request merged**